test/wdio/css-variables/cmp.test.tsx

```diff
@@ -0,0 +1,156 @@
+import './variables.css';
+
+import { Fragment, h } from '@stencil/core';
+import { render } from '@wdio/browser-runner/stencil';
+
+const css = `:root {
+  --global-background: black;
+  --global-color: white;
+  --body-background: grey;
+}
+body {
+  background: var(--body-background);
+  color: var(--global-color);
+}`;
+
+describe('css-variables', () => {
+  beforeEach(() => {
+    render({
+      template: () => (
+        <>
+          <style>{css}</style>
+
+          <css-variables-no-encapsulation></css-variables-no-encapsulation>
+          <css-variables-shadow-dom></css-variables-shadow-dom>
+          {/* This second instance will be used to ensure encapsulation between it and the first instance */}
+          <css-variables-shadow-dom></css-variables-shadow-dom>
+        </>
+      ),
+    });
+  });
+
+  describe('css-variables-no-encapsulation', () => {
+    it('uses class-local css variables to set the background and color', async () => {
+      const blackLocalElm = await $('.black-local');
+
+      await expect(blackLocalElm).toHaveText('No encapsulation: Black background');
+      await expect(blackLocalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)',
+        color: 'rgba(255,255,255,1)',
+      });
+    });
+
+    it('uses global css variables to set the background, color, and font weight', async () => {
+      const blackGlobalElm = await $('.black-global');
+
+      await expect(blackGlobalElm).toHaveText('No encapsulation: Black background (global style)');
+      await expect(blackGlobalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(0, 0, 0, 1)',
+        color: 'rgba(255,255,255,1)',
+      });
+
+      const fontWeight = await blackGlobalElm.getCSSProperty('font-weight');
+      expect(fontWeight.value).toEqual(800);
+    });
+
+    it('uses imported css variables to set the background and color', async () => {
+      const yellowGlobalElm = await $('.yellow-global');
+
+      await expect(yellowGlobalElm).toHaveText('No encapsulation: Yellow background (global link)');
+      await expect(yellowGlobalElm).toHaveStyle({
+        background: 'none 0% 0% auto repeat padding-box border-box scroll rgba(255, 255, 0, 1)',
+        color: 'rgba(0,0,0,1)',
+      });
+    });
+  });
+
+  describe('css-variables-shadow-dom', () => {
+    it("doesn't interfere with global variables", async () => {
+      const shadowRoot = document.querySelector('css-variables-shadow-dom').shadowRoot;
+      const globalShadow = await $('>>>.black-global-shadow');
```

Comment:
The `shadowRoot` here is solely for clicking the button.

I tried to do:
```tsx
await $('>>>button').waitForExist()
const button = await $('>>>button');
await button.click();
```

But that didn't seem to work for some reason. LMK if I'm doing something wrong when it comes to piercing the shadow DOM, as it doesn't appear the click was actually happening.